docs/csharp/language-reference/keywords/readonly.md

```diff
@@ -88,6 +89,51 @@ public readonly struct Point
 
 Adding a field not marked `readonly` generates compiler error `CS8340`: "Instance fields of readonly structs must be readonly."
 
+## Readonly member examples
+
+Other times, you may create a struct that supports mutation. In those cases, several of the instance members likely won't modify the internal state of the struct. You can declare those instance members with the `readonly` modifier. The compiler enforces your intent. If that member modifies state directly, or accesses a member that isn't also declared with the `readonly` modifier, the result is a compile time error. The `readonly` modifier is valid on `struct` members, not `class` or `interface` member declarations.
+
+You gain two advantages by applying the `readonly` modifier to applicable `struct` methods. Most importantly, the compiler enforces your intent. Code that modifies state isn't valid in a `readonly` method. The compiler may also make use of the `readonly` modifier to enable performance optimizations. When large `struct` types are passed by `in` reference, the compiler must generate a defensive copy if the state of the struct might be modified. If only `readonly` members are accessed, the compiler may not create the defensive copy.
+
+The `readonly` modifier is valid on most members of a `struct`, including methods that override methods declared in <xref:System.Object?displayProperty=nameWithType>. There are some restrictions:
+
+- You cannot declare `readonly` static members.
+- You cannot declare `readonly` constructors.
+
+You can add the `readonly` modifier to a property or indexer declaration:
+
+```csharp
+readonly public int Counter
+{
+  get { return 0; }
+  set {} // not useful, but legal
+}
+```
+
+You may also add the `readonly` modifier to individual `get` or `set` accessors of a property or indexer:
+
+```csharp
+public int Counter
+{
+  readonly get { return _counter; }
+  set { _counter = value; }
+}
+int _counter;
+```
+
+You may not add the `readonly` modifier to both a property and one or more of that same property's accessors. That same restriction applies to indexers.
+
+The compiler implicitly applies the `readonly` modifier to auto-implemented properties where the compiler implemented code does not modify state. It is equivalent to the following declarations:
+
+```csharp
+public readonly int Index { get; }
+// Or:
+public int Number { readonly get; }
+public string Message { readonly get; set; }
+``` 
+
+You may add the `readonly` modifier in those locations, but it will have no meaningful effect. You may not add the `readonly` modifier to an auto-implemented property setter, or to a read / write auto-implemented property.
```

Comment:
```suggestion
You may add the `readonly` modifier in those locations, but it will have no meaningful effect. You may not add the `readonly` modifier to an auto-implemented property setter, or to a read/write auto-implemented property.
```